choose-index.md

```diff
@@ -29,19 +29,79 @@ Before introducing index selection, it is important to understand the ways TiDB
 
 ## Index selection rules
 
-TiDB provides a heuristic rule named skyline-pruning based on the cost estimation of each operator for accessing tables. It can reduce the probability of wrong index selection caused by wrong estimation.
+TiDB selects indexes based on rules or cost. The based rules include pre rules and skyline-pruning. When selecting an index, TiDB tries the pre rule first. If an index satisfies a pre rule, TiDB directly selects the index. Otherwise, TiDB uses skyline-pruning to exclude unsuitable indexes, and then based on the cost estimation of each operator for accessing tables, selects the index with the lowest cost.
+
+### Selection based on rules
+
+#### Pre rules
+
+TiDB uses the following heuristic pre rules to select indexes:
+
++ rule 1: If an index satisfies "unique index with full match + no need to retrieve rows from a table (which means the plan generated by the index is IndexReader operator)", directly select this index.
+
++ rule 2: If an index satisfies "unique index + need to retrieve rows from a table (which means the plan generated by the index is IndexReader operator)", select the index with the smallest number of rows to be retrieved from a table as the candidate index.
+
++ rule 3: If an index satisfies "common index + no need to retrieve rows from a table + the number of rows to be read is less than the value of a certain threshold", select the index with the smallest number of rows to be read as the candidate index.
+
++ rule 4: If only one candidate index is selected based on rule 2 and 3, select this index. If two candidate indexes are separately selected based on rule 2 and 3, select the index with the smaller number of rows to be read (the number of rows with index + the number of rows to be retrieved from a table).
+
+The "index with full match" in the above rules means each indexed column has the equal condition. When executing the `EXPLAIN FORMAT = 'verbose' ...` statement, if the pre rules matches an index, TiDB will output a NOTE level warning indicating that the index matches the pre rule.
+
+In the following example, because the index `idx_b` meets the condition "unique index + need to retrieve rows from a table" in rule 2, TiDB selects the index `idx_b` as the access path, and `SHOW WARNING` returns a note indicating that the index  `idx_b` matches the pre rule.
```

Comment:
```suggestion
In the following example, because the index `idx_b` meets the condition "unique index + the need to retrieve rows from a table" in rule 2, TiDB selects the index `idx_b` as the access path, and `SHOW WARNING` returns a note indicating that the index  `idx_b` matches the pre-rule.
```